src/extension/kernel.ts

```diff
@@ -85,6 +86,7 @@ export class Kernel implements Disposable {
   protected runnerReadyListener?: Disposable
 
   protected cellManager = new NotebookCellManager(this.#controller)
+  protected activeTerminals: ActiveTerminal[] = []
```

Comment:
```suggestion
  protected activeTerminals = new WeakMap<string, RunmeTerminal>()
```

This might be better as a `WeakMap`, stored on current execution id. Since currently this list prevents any referenced terminals from being garbage collected, it is a potential memory leak, which WeakMap solves.